pinecone/test_suite.go

```diff
@@ -96,50 +102,58 @@ func GenerateTestIndexName() string {
 }
 
 func upsertVectors(ts *IntegrationTests, ctx context.Context, vectors []*Vector) error {
-	maxRetries := 12
-	delay := 12 * time.Second
-	fmt.Printf("Attempting to upsert vectors into host \"%s\"...\n", ts.host)
-	for i := 0; i < maxRetries; i++ {
-		ready, err := GetIndexStatus(ts, ctx)
-		if err != nil {
-			fmt.Printf("Error getting index ready: %v\n", err)
-			return err
-		}
-		if ready {
-			upsertVectors, err := ts.idxConn.UpsertVectors(ctx, vectors)
-			require.NoError(ts.T(), err)
-			fmt.Printf("Upserted vectors: %v into host: %s\n", upsertVectors, ts.host)
-			break
-		} else {
-			time.Sleep(delay)
-			fmt.Printf("Host \"%s\" not ready for upserting yet, retrying... (%d/%d)\n", ts.host, i, maxRetries)
-		}
+	_, err := WaitUntilIndexReady(ts, ctx)
+	require.NoError(ts.T(), err)
+
+	upsertVectors, err := ts.idxConn.UpsertVectors(ctx, vectors)
+	if err != nil {
+		buf := make([]byte, 1<<16)
+		runtime.Stack(buf, true)
```

Comment:
What does this do ooc?

Reply:
Ah, it prints details of the stack trace on error to the console. I was trying to troubleshoot why we were seeing that content-type error sometimes when trying to upsert to indexes that weren't ready.